### PR TITLE
test: additional integration tests for vpc

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

- Extending test coverage for VPC 

## ✔️ How to Test

1. Setup API token for alpha/beta environment and export it LINODE_TOKEN=mytoken

2. get cacert.pem (e.g. wget https://certurl.com/cacert.pem)

3. make slight modification to `def get_client()` in conftest.py
e.g. `client = LinodeClient(token, base_url='https://api.dev.linode.com/v4beta', ca_path='/Users/ykim/linode/ykim/linode_api4-python/cacert.pem')`

4. run test `pytest test/integration/models/test_vpc.py`

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

File: test/integration/conftest.py [hunk 1]
```diff
@@ -290,3 +290,46 @@ def create_vpc_with_subnet_and_linode(get_client, create_vpc_with_subnet):
     yield vpc, subnet, instance, password
 
     instance.delete()
+
+
+@pytest.fixture(scope="session")
+def create_vpc(get_client):
+    client = get_client
+
+    timestamp = str(int(time.time()))
```
Comment (zliang-akamai) [inline test/integration/conftest.py:299]: Let's use the nanosecond (`str(time.time_ns())`) to guarantee the label is truly unique.
And can we make this as a fixture similar to [this](https://github.com/linode/linodego/blob/proj/vpc/test/integration/util_test.go#L79-L81) for providing a unique label?

File: test/integration/models/test_vpc.py [hunk 2]
```diff
@@ -42,3 +41,69 @@ def test_update_subnet(get_client, create_vpc_with_subnet):
...
+
+    invalid_label = "invalid!!"
+
+    try:
+        vpc.label = invalid_label
+        vpc.save()
+    except ApiError as e:
+        assert e.status == 400
+        assert (
+            "Label must include only ASCII letters, numbers, and dashes"
```
Comment (yec-akamai) [inline test/integration/models/test_vpc.py:81]: I'm a bit worried that asserting the exact error message may lead to test failure in the future, if the returned error message is just slightly changed on the API side. What do you think?

**Comment by yec-akamai** [comment]
Can you add more details on how to run the integration tests?

File: test/integration/models/test_vpc.py (hunk 2 above)
Comment (ykim-1) [reply test/integration/models/test_vpc.py:81]: Hm, yeah that could be the case I can make the assertion just on keywords

**Comment by ykim-1** [comment]
> Nice work! There is probably some API issue ongoing for the subnet related endpoints in alpha. I ran the tests and the subnet test cases failed due to ApiError: 500: Service unavailable.

I will try to run it again tmrw to make sure

File: test/integration/models/test_vpc.py [hunk 3]
```diff
@@ -42,3 +41,57 @@ def test_update_subnet(get_client, create_vpc_with_subnet):
...
+def test_fails_create_vpc_invalid_data(get_client):
+    try:
+        get_client.vpcs.create(
+            label="invalid_label!!",
+            region=get_region(get_client, {"VPCs"}),
+            description="test description",
+        )
+    except ApiError as e:
+        assert e.status == 400
+        assert "Label must include only ASCII" in str(e.json)
```
Comment (zliang-akamai) [inline test/integration/models/test_vpc.py:55]:
```suggestion
    with pytest.raises(ApiError) as excinfo:
        get_client.vpcs.create(
            label="invalid_label!!",
            region=get_region(get_client, {"VPCs"}),
            description="test description",
        )
    assert e.status == 400
    assert "Label must include only ASCII" in str(e.json)
```

Can we use the `pytest.raises` to ensure the exception is raised?

Comment (ykim-1) [reply test/integration/models/test_vpc.py:55]: Looks cleaner this way. I will make the change

File: test/integration/conftest.py [hunk 4]
```diff
@@ -300,3 +301,46 @@ def create_vpc_with_subnet_and_linode(get_client, create_vpc_with_subnet):
     yield vpc, subnet, instance, password
 
     instance.delete()
+
+
+@pytest.fixture(scope="session")
```
Comment (zliang-akamai) [inline test/integration/conftest.py:306]: I think the scope is not session? The new VPC will need to be created every time this fixture is accessed, right? Or not?

Comment (ykim-1) [reply test/integration/conftest.py:306]: Test cases that exist right now can work with this one session vpc create instance. We can change this as we add more tests that require different instances to be created

**Review by zliang-akamai — APPROVED** [review]
Tested locally and everything worked well! 👍

**Pull request merged**